ui/js/directives/perf/compare.js

```diff
@@ -45,6 +54,7 @@ treeherder.directive(
                             scope.filteredResultList[key] = compareResults;
                         }
                     });
+                    scope.filteredResultList = sortResults(scope.filteredResultList);
```

Comment:
Oh right, I remember this happening to me before elsewhere in treeherder! It's a bit tricky. I'm not sure if your solution will work as you're still returning an object? Maybe I'm missing something.

Something like this makes a bit more sense to me:
```
 _.map(_.keys(scope.filteredResultList), function(testName) { return { 'testName': testName, 'results': scope.filteredResultList[testName] } })
```

Then you can use orderBy and just iterate through it like an array (accessing the testName and results properties as needed from each element)